File: web/src/main/java/top/bettercode/summer/web/ApplicationContextHolderConfiguration.java
package top.bettercode.summer.web;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import top.bettercode.summer.web.support.ApplicationContextHolder;

@Configuration(proxyBeanMethods = false)
public class ApplicationContextHolderConfiguration {

  @Bean
  public ApplicationContextHolder applicationContextHolder() {
    return new ApplicationContextHolder();
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/config/ApplicationContextHolderConfiguration.kt
package top.bettercode.summer.web.config

import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Configuration
import org.springframework.core.Ordered
import org.springframework.core.annotation.Order
import top.bettercode.summer.web.support.ApplicationContextHolder

@Order(Ordered.HIGHEST_PRECEDENCE)
@Configuration(proxyBeanMethods = false)
class ApplicationContextHolderConfiguration(applicationContext: ApplicationContext) {

    init {
        ApplicationContextHolder.applicationContext = applicationContext
    }
}